test/framework/ssh/ssh.go

```diff
@@ -48,6 +50,20 @@ func NewApp(verbose bool, port string, envMap map[string]string, args []string)
 	inWriters := []io.Reader{&app.stdin}
 	outWriters := []io.Writer{&app.stdout}
 	errWriters := []io.Writer{&app.stderr}
+
+	logDirName := "/tmp/e2e"
+	err := os.MkdirAll(logDirName, os.ModePerm)
+	if err != nil {
+		panic(errors.Wrap(err, "could not create /tmp/e2e"))
+	}
+	logFileName := logDirName + "/" + appName
+	f, err := os.OpenFile(logFileName, os.O_RDWR|os.O_APPEND|os.O_CREATE, 0660)
+	if err != nil {
+		panic(errors.Wrap(err, "could not create "+logFileName))
+	}
```

Comment:
Any reason not to use the Ginkgo writers? There's a TeeTo option that could help: https://onsi.github.io/ginkgo/#logging-output

I think the advantage of this is that it'd be in circleci but also if configured correctly in test-reports.

Reply:
```
❯❯❯ cat /tmp/e2e/ginkgo
STEP: installing one cross-mesh gateway and one non-cross-mesh gateway 07/29/22 12:00:56.945
STEP: demo-client specific RateLimit works 07/29/22 12:01:04.125
STEP: catch-all RateLimit works 07/29/22 12:01:06.208
STEP: Exposing external service and specific rate limit 07/29/22 12:01:07.012
STEP: demo-client specific RateLimit works 07/29/22 12:01:07.106
STEP: RateLimit doesn't apply for other clients 07/29/22 12:01:08.678
STEP: Checking requests succeed 07/29/22 12:01:09.808
STEP: Adding a faulty dataplane 07/29/22 12:01:12.035
STEP: Check some errors happen 07/29/22 12:01:12.091
STEP: Checking requests succeed 07/29/22 12:01:21.012
STEP: check requests take over 5s 07/29/22 12:01:29.299
STEP: apply a new policy 07/29/22 12:01:34.41
STEP: eventually requests timeout consistently 07/29/22 12:01:34.449
STEP: Apply a retry policy 07/29/22 12:03:17.775
STEP: Eventually all requests succeed consistently 07/29/22 12:03:17.876
```
that's the result of `TeeTo` when running `make -j test/e2e-universal`.

Even if it would dump all the stdout of the test, I still think that separate files for containers are much better for debugging the test.